src/common/hooks/useAuth/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { roleAtom, setTokenAtom, tokenAtom, UserRole } from 'atoms/authAtom';
+import { useAtom } from 'jotai';
+
+import { Navigate, useNavigate } from "react-router-dom";
+import { AuthResult } from './types';
+import { useAuthLogin, useAuthLogout } from 'queries/modules/auth/useAuthQuery';
+import { removeTokenFromLocalStorage } from 'utils/authStorage';
+import { LoginRequest } from 'api/modules/auth/types';
+
+/**
+ *
+ * @param requiredRoles // 관리자 Role
+ * @param redirectPath // 리다이렉트 경로
+ * @returns {AuthResult}
+ * @description
+ * 인증 관련 Hook
+ * - requiredRoles: 해당 페이지에 접근하기 위해 필요한 Role
+ * - redirectPath: 인증되지 않은 사용자가 접근 시 리다이렉트 될 경로
+ * - isAuthenticated: 인증된 사용자인지 여부
+ * - AuthGuard: 인증된 사용자만 접근 가능한 페이지를 위한 컴포넌트
+ * - login: 로그인 함수
+ * - logout: 로그아웃 함수
+ */
+
+const useAuth = (requiredRoles: UserRole[], redirectPath: string = "/sign-in"): AuthResult => {
```

Comment:
default parameter를 사용하는 경우에는 자동으로 추론이 되어서 제거해줘도 괜찮을 것 같네요!
```suggestion
const useAuth = (requiredRoles: UserRole[], redirectPath = "/sign-in"): AuthResult => {
```

Reply:
리턴 값도 추론되는줄은 몰랐네요. 감사합니다. 저도 타입 추론에 관해서는 추론을 쫒아가는편이에요.